libraries/shared/src/RegisteredMetaTypes.cpp

```diff
@@ -46,81 +44,6 @@ int voidLambdaType = qRegisterMetaType<std::function<void()>>();
 int variantLambdaType = qRegisterMetaType<std::function<QVariant()>>();
 int stencilModeMetaTypeId = qRegisterMetaType<StencilMaskMode>();
```

Comment:
it seems confusing to me that these qRegisterMetaType calls are still here and the variant conversions below

Reply:
qRegisterMetaType is not part of QtScript and is rather part of the Q_DECLARE_METATYPE macro for signal/slot behaviors.

Reply:
sure, but it would maybe make more sense to still keep the qRegisterMetaType calls close to where the conversion methods are defined?

Reply:
qRegisterMetaType has no direct linkage to the script engine, it's a way of informing Qt that this is a type that may be used in a signal or slot.  So having it be associated with the script-engine conversion methods feels slightly irrelevant (unless I'm misunderstanding something here of course)

Reply:
I feel like it's implicitly linked because our signals and slots are very often exposed to the scripting engine (via the meta objects).  e.g. we're registering vec2 here, but defining its conversion to script engine somewhere else.  it'd just be nice to have those close together, so if we ever un-exposed vec2 (or whatever), we just have to change one place

Reply:
If we ever un-exposed vec2 then we may still need the qRegisterMetaType in case we had anything in our code calling a signal/slot through Qt and passed this along.  This class is defined in "shared", not in "script-engine", I can't guarantee that there aren't other Qt uses of this object.